atlasdb-conjure/src/test/java/com/palantir/atlasdb/http/v2/ClientOptionsTest.java

```diff
@@ -0,0 +1,107 @@
+import com.palantir.conjure.java.api.config.service.UserAgent;
+import com.palantir.conjure.java.client.config.ClientConfiguration;
+
+public class ClientOptionsTest {
+    private static final UserAgent USER_AGENT = UserAgent.of(UserAgent.Agent.of("tom", "1.2.3"));
+
+    // Throws after expected outages of 1/2 * 0.01 * (2^13 - 1) = 40.96 s
+    private static final ClientOptions DEFAULT_RETRYING = ImmutableClientOptions.builder()
+            .connectTimeout(Duration.ofMillis(500))
+            .readTimeout(Duration.ofSeconds(65))
+            .backoffSlotSize(Duration.ofMillis(10))
+            .failedUrlCooldown(Duration.ofMillis(100))
+            .maxNumRetries(13)
+            .clientQoS(ClientConfiguration.ClientQoS.DANGEROUS_DISABLE_SYMPATHETIC_CLIENT_QOS)
+            .build();
+
+    private static final ClientOptions DEFAULT_NO_RETRYING = ImmutableClientOptions.builder()
+            .connectTimeout(Duration.ofMillis(500))
+            .readTimeout(Duration.ofSeconds(65))
+            .backoffSlotSize(Duration.ofMillis(10))
+            .failedUrlCooldown(Duration.ofMillis(1))
+            .maxNumRetries(0)
+            .clientQoS(ClientConfiguration.ClientQoS.DANGEROUS_DISABLE_SYMPATHETIC_CLIENT_QOS)
+            .build();
+
+    @Test
+    public void proxyShouldSupportBlockingReadTimeoutIfUnspecified() {
+        ClientOptions clientOptions = ClientOptions.fromRemotingParameters(AuxiliaryRemotingParameters.builder()
+                .shouldLimitPayload(true)
+                .shouldRetry(true)
+                .userAgent(USER_AGENT)
+                .build());
+
+        assertThat(clientOptions.readTimeout()).isEqualTo(ClientOptions.BLOCKING_READ_TIMEOUT);
+    }
+
+    @Test
+    public void proxyShouldSupportBlockingReadTimeoutIfExplicitlyConfigured() {
+        ClientOptions clientOptions = ClientOptions.fromRemotingParameters(AuxiliaryRemotingParameters.builder()
+                .shouldLimitPayload(true)
+                .shouldRetry(true)
+                .shouldSupportBlockingOperations(true)
+                .userAgent(USER_AGENT)
+                .build());
+
+        assertThat(clientOptions.readTimeout()).isEqualTo(ClientOptions.BLOCKING_READ_TIMEOUT);
+    }
+
+    @Test
+    public void proxyShouldSupportNonBlockingReadTimeoutIfExplicitlyConfigured() {
+        ClientOptions clientOptions = ClientOptions.fromRemotingParameters(AuxiliaryRemotingParameters.builder()
+                .shouldLimitPayload(true)
+                .shouldRetry(true)
+                .shouldSupportBlockingOperations(false)
+                .userAgent(USER_AGENT)
+                .build());
+
+        assertThat(clientOptions.readTimeout()).isEqualTo(ClientOptions.NON_BLOCKING_READ_TIMEOUT);
+    }
+
+    @Test
+    public void defaultRetryingOptionsShouldMatchLegacyBehaviour() {
+        ClientOptions clientOptions = ClientOptions.fromRemotingParameters(AuxiliaryRemotingParameters.builder()
```

Comment:
this test is kinda weird, there's no notion of default `AuxiliaryRemotingParameters`

Reply:
Sure, yep. I changed this to `minimallySpecified...` as that's what I meant here